src/main/java/com/spotify/docker/client/auth/ConfigFileRegistryAuthSupplier.java

```diff
@@ -77,8 +77,12 @@ public RegistryAuth authFor(final String imageName) throws DockerException {
       }
       return reader.fromConfig(path, ref.getRegistryName());
     } catch (IllegalArgumentException e) {
-      // no configuration for registry
-      return null;
+      log.debug("Failed first attempt to find auth for {}", ref.getRegistryUrl(), e);
+      try {
+        return reader.fromConfig(path, ref.getRegistryName());
+      } catch (IllegalArgumentException | IOException e2) {
+        throw new DockerException(e2);
```

Comment:
this shouldn't throw an exception, as there are other code paths that call `authFor(image)` and expect a `null` response if the config file / supplier has no authentication info for the given image name. It's not necessarily an error to not have RegistryAuth info for a given image.